### PR TITLE
[chore] use macos-14 for arm testing

### PR DESCRIPTION
**Pull request by codeboten**
Similar to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33921

### CODE REVIEW

**Review by songy23 — APPROVED** [review]

File: .github/workflows/build-and-test-arm.yml [hunk 1]
```diff
@@ -50,7 +49,7 @@ jobs:
         run: make -j4 gotest
   arm-unittest:
     if: ${{ github.actor != 'dependabot[bot]' && (contains(github.event.pull_request.labels.*.name, 'Run ARM') || github.event_name == 'push' || github.event_name == 'merge_group') }}
-    runs-on: actuated-arm64-4cpu-4gb
```
Comment (jpkrohling) [inline .github/workflows/build-and-test-arm.yml:53]: what was this one? are we skipping this in favor of macos-14, or are they equivalent?

**Review by mx-psi — APPROVED** [review]

File: .github/workflows/build-and-test-arm.yml (hunk 1 above)
Comment (codeboten) [reply .github/workflows/build-and-test-arm.yml:53]: the [actuated runners](https://github.com/self-actuated/actuated) were used to test arm only. the macos-14 provides the same coverage AFAIK

**Comment by mx-psi** [comment]
I think after this change what we claim on https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/platform-support.md#tier-2--secondary-support is no longer correct. But realistically, the current arm runners do not provide good test coverage, they are too flaky, so maybe we can just downgrade our support for Linux ARM for now

**Comment by codeboten** [comment]
> I think after this change what we claim on https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/platform-support.md#tier-2--secondary-support is no longer correct. But realistically, the current arm runners do not provide good test coverage, they are too flaky, so maybe we can just downgrade our support for Linux ARM for now

Pinging @atoulme for review on this as well since he's the owner of linux/arm

**Review by TylerHelmuth — APPROVED** [review]

**Comment by codeboten** [comment]
@mx-psi as per your suggestion, i've updated the supported platforms doc to reflect this change

**Comment by atoulme** [comment]
I am acknowledging this change. I didn't get a chance to review and weigh in. I will take further actions to bring back support of linux/arm64 back to tier 2. Thanks.

**Pull request merged**